src/TrajectoryPlanner/src/TrajectoryGenerator.cpp

```diff
@@ -496,6 +542,22 @@ bool TrajectoryGenerator::isTrajectoryAsked()
     return m_generatorState == GeneratorState::Called;
 }
 
+bool TrajectoryGenerator::generateTrajectoriesFromFootprints(std::shared_ptr<FootPrint> left, std::shared_ptr<FootPrint> right,
+                                                             const double &initTime)
+{
+    DCMInitialState initialState;
+    initialState.initialPosition = m_DCMBoundaryConditionAtMergePointPosition;
+    initialState.initialVelocity = m_DCMBoundaryConditionAtMergePointVelocity;
+
+    if (!m_dcmGeneratorStepAdj->setDCMInitialState(initialState))
```

Comment:
From the name of the method, it is not clear at all that you are going to set these footprints to the adjusted version.

Reply:
I changed the name of method but I didn't get your point about footprint. Here we are setting the DCM boundary condition, what do you mean about footprint?

Reply:
Sorry I made a mistake I got your point and I reverted the changed that I did before. I guess you were talking about the previous issue that the we can solve it by making an object of trajectory generator!

Reply:
Done!